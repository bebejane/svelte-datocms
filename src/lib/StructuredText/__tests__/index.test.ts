/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom';
import type {StructuredText as StructuredTextGraphQlResponse, Document} from 'datocms-structured-text-utils'
import { render } from '@testing-library/svelte';
import { StructuredText } from '../../index';

describe('StructuredText', () => {
	describe('with no value', () => {
		it('renders null', () => {
			const { container } = render(StructuredText, { data: null });
			expect(container.querySelector('pre').textContent).toBe('No data document')
			//expect(component).toMatchSnapshot();
		});
	});
});
describe('simple dast /2', () => {
	const structuredText: Document = {
		schema: 'dast',
		document: {
			type: 'root',
			children: [
				{
					type: 'heading',
					level: 1,
					children: [
						{
							type: 'span',
							value: 'This\nis a '
						},
						{
							type: 'span',
							marks: ['strong'],
							value: 'title'
						}
					]
				}
			]
		}
	};
	describe('with default rules', () => {
		it('renders the document', () => {
			const { container } = render(StructuredText, { data: structuredText });
			
			expect(container).toMatchSnapshot();
		});
	});
});
describe("simple dast with no links/blocks", () => {
    const structuredText: StructuredTextGraphQlResponse = {
      value: {
        schema: "dast",
        document: {
          type: "root",
          children: [
            {
              type: "heading",
              level: 1,
              children: [
                {
                  type: "span",
                  value: "This\nis a ",
                },
                {
                  type: "span",
                  marks: ["strong"],
                  value: "title",
                },
              ],
            },
          ],
        },
      },
    };

    describe("with default rules", () => {
      it("renders the document", () => {
        const { container } = render(StructuredText, { data: structuredText });
        expect(container).toMatchSnapshot();
      });
    })})
	describe("with links/blocks", () => {
		type QuoteRecord = {
		  id: string;
		  __typename: "QuoteRecord";
		  quote: string;
		  author: string;
		};
	
		type DocPageRecord = {
		  id: string;
		  __typename: "DocPageRecord";
		  slug: string;
		  title: string;
		};
	
		const structuredText: StructuredTextGraphQlResponse<
		  QuoteRecord | DocPageRecord
		> = {
		  value: {
			schema: "dast",
			document: {
			  type: "root",
			  children: [
				{
				  type: "heading",
				  level: 1,
				  children: [
					{
					  type: "span",
					  value: "This is a",
					},
					{
					  type: "span",
					  marks: ["highlight"],
					  value: "title",
					},
					{
					  type: "inlineItem",
					  item: "123",
					},
					{
					  type: "itemLink",
					  item: "123",
					  meta: [
						{ id: 'target', value: '_blank' },
					  ],
					  children: [{ type: "span", value: "here!" }],
					},
				  ],
				},
				{
					type:'paragraph',
					children:[
						{
							type: 'link',
							url: 'http://www.datocms.com',
							"meta" : [
								{id: 'target', value:'_blank'},
								{id: 'rel', value: 'nofollow'},
							],
							children: [
								{type:'span', value:'Empty link'}
							]
						},
					]
				},
				{
				  type: "block",
				  item: "456",
				},
			  ],
			},
		  },
		  blocks: [
			{
			  id: "456",
			  __typename: "QuoteRecord",
			  quote: "Foo bar.",
			  author: "Mark Smith",
			},
		  ],
		  links: [
			{
			  id: "123",
			  __typename: "DocPageRecord",
			  title: "How to code",
			  slug: "how-to-code",
			},
		  ],
		};

		describe("with default rules", () => {
			it("renders the document", () => {
				const { container } = render(StructuredText, { data: structuredText });
        		expect(container).toMatchSnapshot();
			})
		})
	})