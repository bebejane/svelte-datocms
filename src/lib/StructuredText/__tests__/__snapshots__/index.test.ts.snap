// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple dast /2 with default rules renders the document 1`] = `
StructuredText {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      Object {
        "document": Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "span",
                  "value": "This
is a ",
                },
                Object {
                  "marks": Array [],
                  "type": "span",
                  "value": "title",
                },
              ],
              "level": 1,
              "type": "heading",
            },
          ],
          "type": "root",
        },
        "schema": "dast",
      },
      Object {
        "document": Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "type": "span",
                  "value": "This
is a ",
                },
                Object {
                  "marks": Array [],
                  "type": "span",
                  "value": "title",
                },
              ],
              "level": 1,
              "type": "heading",
            },
          ],
          "type": "root",
        },
        "schema": "dast",
      },
      undefined,
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "components": 2,
      "data": 0,
    },
    "root": <div>
      <div>
        <h1
          class="text-6xl"
        >
          This
is a 
          
          
          
          
          <strong>
            title
            
             
          </strong>
          
          
          
          
          
          
          
        </h1>
        
        
        
        
        
      </div>
      
      
      
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
